File: parking-domain/src/main/java/com/chatchatabc/parking/impl/infra/service/KVServiceRedisImpl.java
package com.chatchatabc.parking.impl.infra.service;

import com.chatchatabc.parking.infra.service.KVService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.redis.core.RedisTemplate;
import org.springframework.data.redis.core.ValueOperations;
import org.springframework.stereotype.Service;

import java.util.concurrent.TimeUnit;

@Service
public class KVServiceRedisImpl implements KVService {
    @Autowired
    private RedisTemplate<String, String> redisTemplate;

    /**
     * Set key value pair
     *
     * @param key   the key
     * @param value the value
     * @param ttl   the time to live in seconds
     */
    @Override
    public void set(String key, String value, Long ttl) {
        ValueOperations<String, String> valueOperations = redisTemplate.opsForValue();
        if (ttl != null) {
            valueOperations.set(key, value, ttl, TimeUnit.SECONDS);
        } else {
            valueOperations.set(key, value);
        }
    }

    /**
     * Get value by key
     *
     * @param key the key
     * @return the value
     */
    @Override
    public String get(String key) {
        ValueOperations<String, String> valueOperations = redisTemplate.opsForValue();
        return valueOperations.get(key);
    }

    /**
     * Delete value by key
     *
     * @param key the key
     */
    @Override
    public void delete(String key) {
        redisTemplate.delete(key);
    }
}

File: parking-api/src/main/kotlin/com/chatchatabc/parking/api/impl/application/service/KVServiceRedisImpl.kt
package com.chatchatabc.parking.api.impl.application.service

import com.chatchatabc.parking.api.application.service.KVService
import org.springframework.data.redis.core.RedisTemplate
import org.springframework.stereotype.Service
import java.util.concurrent.TimeUnit

@Service
class KVServiceRedisImpl(
    private val redisTemplate: RedisTemplate<String, String>
) : KVService {

    /**
     * Set key value pair
     */
    override fun set(key: String, value: String, ttl: Long?) {
        val ops = redisTemplate.opsForValue()
        if (ttl != null) {
            ops.set(key, value, ttl, TimeUnit.SECONDS)
        } else {
            ops.set(key, value)
        }
    }

    /**
     * Get value by key
     */
    override fun get(key: String): String? {
        val ops = redisTemplate.opsForValue()
        return ops.get(key)
    }

    /**
     * Delete key value pair
     */
    override fun delete(key: String) {
        redisTemplate.delete(key)
    }
}